src/types/media-session.ts

```diff
@@ -103,3 +85,19 @@ export class GenesysCloudMediaSession extends MediaSession {
     });
   }
 }
+
+export interface SessionEvents {
+  iceConnectionType: ({localCandidateType: string, relayed: boolean, remoteCandidateType: string});
+  peerTrackAdded: (track: MediaStreamTrack, stream?: MediaStream) => void;
+  peerTrackRemoved: (track: MediaStreamTrack, stream?: MediaStream) => void;
+  mute: JingleInfo;
+  unmute: JingleInfo;
+  sessionState: 'starting' | 'pending' | 'active';
+  connectionState: 'starting' | 'connecting' | 'connected' | 'interrupted' | 'disconnected' | 'failed';
+  terminated: JingleReason;
+  stats: StatsEvent;
+  endOfCandidates: void;
+}
+
+applyMixins(GenesysCloudMediaSession, [ EventEmitter ]);
+export interface GenesysCloudMediaSession extends StrictEventEmitter<EventEmitter, SessionEvents> { }
```

Comment:
These two lines are derived from the typescript cookbook. basically it is a way to do multiple inheritance. Since we are already extending jingle's `MediaSession` and we *also* was to extend/add an `EventEmitter` we basically specify it in the interface and then do a javascripty way of extending the class at runtime.